src/anomalib/models/image/uninet/lightning_model.py

```diff
@@ -0,0 +1,174 @@
+"""Lightning model for UniNet."""
+
+# Copyright (C) 2025 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import copy
+
+import torch
+import torchvision
+from lightning.pytorch.utilities.types import STEP_OUTPUT
+from torch import nn
+from torch.nn import functional
+from torchvision.models.feature_extraction import create_feature_extractor
+
+from anomalib.data import Batch, InferenceBatch
+from anomalib.models.components import AnomalibModule
+from anomalib.models.components.backbone import get_decoder
+
+from .anomaly_map import weighted_decision_mechanism
+from .components import (
+    AttentionBottleneck,
+    BottleneckLayer,
+    DomainRelatedFeatureSelection,
+    domain_related_feature_selection,
+)
+from .loss import UniNetLoss
+from .torch_model import UniNetModel
+
+
+class UniNet(AnomalibModule):
+    """UniNet model for anomaly detection.
+
+    Args:
+        student_backbone (str): The backbone model to use for the student.
+        teacher_backbone (str): The backbone model to use for the teacher.
+    """
+
+    def __init__(
+        self,
+        student_backbone: str = "wide_resent50_2",
+        teacher_backbone: str = "wide_resent50_2",
+    ) -> None:
+        self.dfs = DomainRelatedFeatureSelection()
+        self.loss = UniNetLoss(temperature=2.0)
+        self.bce_loss = torch.nn.BCEWithLogitsLoss()
+        source_teacher = self._get_teacher(teacher_backbone)
+        target_teacher = copy.deepcopy(source_teacher)
+        self.model = UniNetModel(
+            student=get_decoder(student_backbone),
+            bottleneck=BottleneckLayer(block=AttentionBottleneck, layers=3),
+            source_teacher=source_teacher,
+            target_teacher=target_teacher,
+        )
+
+    def training_step(self, batch: Batch, *args, **kwargs) -> STEP_OUTPUT:
+        """Perform a training step of UniNet."""
+        del args, kwargs  # These variables are not used.
+        source_target_features, student_features, predictions = self.model(batch.image)
+        student_features = self._feature_selection(source_target_features, student_features)
+        loss = self._compute_loss(student_features, source_target_features, predictions, batch.gt_label, batch.gt_mask)
```

Comment:
would it be possible if torch model directly returns loss in training step and predictions in eval steps?